src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -309,73 +309,73 @@ private static void CollectProperties(Dictionary<ObjectTypeProperty, CodeWriterD
             }
         }
 
-        private static void DeserializeIntoVariable(this CodeWriter writer, JsonSerialization serialization, Action<CodeWriter, CodeWriterDelegate> valueCallback,
-            CodeWriterDelegate element, bool? writeNullHandling = null)
+        private static void DeserializeIntoVariableWithNullHandling(this CodeWriter writer, JsonSerialization serialization, Action<CodeWriter, CodeWriterDelegate> valueCallback,
+            CodeWriterDelegate element, bool? writeNullHandling)
         {
-            void WriteNullableScope(Action writeContent)
+            if (serialization is JsonArraySerialization || serialization is JsonDictionarySerialization)
             {
-                if (writeNullHandling == true)
+                writeNullHandling ??= true;
```

Comment:
Doesn't type based logic work here too? JsonArraySerialization.Type would be a reference type.

Reply:
👀 Right!

Reply:
Unfortunately, you still need a mechanism to do the type check since you need the actual type to get the `Type` property. `JsonSerialization` base type has no `Type` property.

Reply:
Done.